frame/utility/src/lib.rs

```diff
@@ -145,7 +141,8 @@ decl_module! {
 		#[weight = (
 			calls.iter()
 				.map(|call| call.get_dispatch_info().weight)
-				.fold(15_000_000, |a: Weight, n| a.saturating_add(n).saturating_add(1_000_000)),
+				.sum::<Weight>()
```

Comment:
I'm not sure about this sum, isn't it a wrapping add ?
In some extirnsic user can give an upper on some size of on-chain stuff, user could then just make upper bound being very high, this sum would wrap around and final weight will be just the add below

Reply:
This is a good point. I did not consider this